skimage/measure/segmentation_metrics.py

```diff
@@ -0,0 +1,315 @@
+
+    Returns
+    -------
+    are : float
+        The adapted Rand error; equal to $1 - \frac{2pr}{p + r}$,
+        where $p$ and $r$ are the precision and recall described below.
+    prec : float
+        The adapted Rand precision.
+    rec : float
+        The adapted Rand recall.
+
+    References
+    ----------
+    [1]: http://brainiac2.mit.edu/SNEMI3D/evaluation
+    """
+    _assert_compatible(im_true, im_test)
+
+    # segA is query, segB is truth
+    segA = im_test
+    segB = im_true
+
+    n = segA.size
+
+    # This is the contingency table obtained from segA and segB, we obtain
+    # the marginal probabilities from the table.
+    p_ij = _contingency_table(segB, segA)
+
+    # Sum of the joint distribution squared
+    sum_p_ij = p_ij.data @ p_ij.data
+
+    # These are the axix-wise sums (np.sumaxis)
+    a_i = p_ij.sum(axis=0).A.ravel()
+    b_i = p_ij.sum(axis=1).A.ravel()
+
+    # Sum of the segment labeled 'A'
+    sum_a = a_i @ a_i
+    # Sum of the segment labeled 'B'
+    sum_b = b_i @ b_i
+
+    precision = (sum_p_ij - n)/ (sum_a - n)
+    recall = (sum_p_ij - n)/ (sum_b - n)
+
+    fscore = 2. * precision * recall / (precision + recall)
+    are = 1. - fscore
+
+    return (are, precision, recall)
+
+def compare_variation_of_information(im1, im2, weights=np.ones(2)):
+    """Return the variation of information metric. [1]
+
+    VI(X, Y) = H(X | Y) + H(Y | X), where H(.|.) denotes the conditional
+    entropy.
+
+    Parameters
+    ----------
+    im1, im2 : ndarray
+        Image.  Any dimensionality.
+    weights : ndarray of float, shape (2,), optional
+        The weights of the conditional entropies of `im1` and `im2`. Equal weights
+        are the default.
+
+    Returns
+    -------
+    v : float
+        The variation of information between `im1` and `im2`.
+
+    References
+    ----------
+    [1] Meila, M. (2007). Comparing clusterings - an information based
+    distance. Journal of Multivariate Analysis 98, 873-895.
+    """
+    return np.dot(weights, compare_split_variation_of_information(im1, im2))
+
+def compare_split_variation_of_information(im1, im2):
+    """Return the symmetric conditional entropies associated with the VI.
+
+    The variation of information is defined as VI(X,Y) = H(X|Y) + H(Y|X).
+    If Y is the ground-truth segmentation, then H(Y|X) can be interpreted
+    as the amount of under-segmentation of Y and H(X|Y) is then the amount
+    of over-segmentation.  In other words, a perfect over-segmentation
+    will have H(Y|X)=0 and a perfect under-segmentation will have H(X|Y)=0.
+
+    Parameters
+    ----------
+    im1, im2 : ndarray
+        Image.  Any dimensionality.
+
+    Returns
+    -------
+    sv : ndarray of float, shape (2,)
+        The conditional entropies of im2|im1 and im1|im2.
+
+    See Also
+    --------
+    vi
+    """
+    _, _, _ , hxgy, hygx, _, _ = _vi_tables(im1, im2)
+    # false merges, false splits
+    return np.array([hygx.sum(), hxgy.sum()])
```

Comment:
Depending on the order of `im_true` and `im_test`, you might need to reorder "false merges, false splits". You can test this with the simple "images": `[1, 1]`, `[1, 2]`. (The second one is "split" relative to the first. So depending on which is `im_true`, you have either a false split and no false merges, or a false merge and no false splits.